### PR TITLE
changes to reflect latest community design

### LINKED ISSUE #27: Update smart contract to reflect latest community design
**Acceptance Criteria**

- Remove labels field, add tag and handle. There should be only one tag and only one handle
- Add sponsorship boolean. (For frontend) When this set to true, sponsorship type of posts of community's label should be post
- Remove overview_page_markdown and events_page_markdown, add wiki1 and wiki2. Each of the wiki should be optional. When it's turned on, it has a name, e.g. "Overview", and markdown content.
- Telegram handles should be only one
- Deploy a version of the updated contract to testnet, and provide test commands in near-cli

### PR DESCRIPTION
**Pull request by ailisp**
Resolves #27 .

For @near-akaia-root: I deployed a version to testnet, you can interact with it with near-cli using these commands:
```sh


export contract=m.zxcvn.testnet

near call $contract add_community --accountId k.zxcvn.testnet --args '{"handle":"zkp2", "community":{"name":"zero knowledge", "description":"a community discussing zkp on NEAR", "image_url":"https://ipfs.near.social/ipfs/bafkreihgxg5kwts2juldaeasveyuddkm6tcabmrat2aaq5u6uyljtyt7lu", "thumbnail_url":"https://ipfs.near.social/ipfs/bafkreiajwq6ep3n7veddozji2djv5vviyisabhycbweslvpwhsoyuzcwi4","wiki1":{"name":"overview", "content_markdown":"Zero Knowledge (ZK) technology is gaining significant traction within the NEAR community. This space brings together individuals, experts, and organizations who are interested in developing a ZK ecosystem on NEAR. Our vision is to drive the development of ZK technology and explore its potential applications.Our GroupsThe Zero Knowledge community consists of two main groups:..."},"wiki2":null, "admins":[], "tag":"zero-knowledge","telegram_handle":null,"github":null,"sponsorship":true}}'
near call $contract edit_community --accountId k.zxcvn.testnet --args '{"handle":"zkp2", "community":{"name":"zero knowledge 2", "description":"a community discussing zkp on NEAR", "image_url":"https://ipfs.near.social/ipfs/bafkreihgxg5kwts2juldaeasveyuddkm6tcabmrat2aaq5u6uyljtyt7lu", "thumbnail_url":"https://ipfs.near.social/ipfs/bafkreiajwq6ep3n7veddozji2djv5vviyisabhycbweslvpwhsoyuzcwi4","wiki1":{"name":"overview", "content_markdown":"Zero Knowledge (ZK) technology is gaining significant traction within the NEAR community. This space brings together individuals, experts, and organizations who are interested in developing a ZK ecosystem on NEAR. Our vision is to drive the development of ZK technology and explore its potential applications.Our GroupsThe Zero Knowledge community consists of two main groups:..."},"wiki2":null, "admins":[], "tag":"zero-knowledge","telegram_handle":null,"github":null,"sponsorship":true}}'
near view $contract get_all_communities
near view $contract get_community '{"handle":"zkp2"}'
```

### CODE REVIEW

File: src/community.rs [hunk 1]
```diff
@@ -2,26 +2,34 @@ use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
...
     pub thumbnail_url: String,
-    pub overview_page_markdown: String,
-    pub events_page_markdown: String,
+    pub wiki1: Option<Wiki>,
+    pub wiki2: Option<Wiki>,
     pub admins: Vec<AccountId>,
-    pub labels: Vec<String>,
-    pub telegram_handles: Vec<String>,
+    pub tag: String,
+    pub telegram_handle: Option<String>,
```
Comment (frol) [inline src/community.rs:23]: Do we have any communities in the contract? If so, this is going to be a state-breaking-change

File: src/community.rs [hunk 2]
```diff
@@ -2,26 +2,34 @@ use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
...
     pub image_url: String,
     pub thumbnail_url: String,
-    pub overview_page_markdown: String,
-    pub events_page_markdown: String,
+    pub wiki1: Option<Wiki>,
+    pub wiki2: Option<Wiki>,
     pub admins: Vec<AccountId>,
-    pub labels: Vec<String>,
-    pub telegram_handles: Vec<String>,
+    pub tag: String,
```
Comment (frol) [inline src/community.rs:22]: How `Community::tag` is different from `CommunityCard::handle`? Is it the same thing? It is confusing, please, either rename it to be the same or add comments in the code 🙏

**Review by frol — APPROVED** [review]
Approving to unblock; see some concerns in the comments

**Comment by akaia-shadowfox** [comment]
Thank you very much @ailisp!

File: src/community.rs (hunk 2 above)
Comment (ori-near) [reply src/community.rs:22]: @frol – I think we were thinking of renaming "labels" to "tags" to make it consistent with NEAR Social conventions. And handle would be the URL slug.

Example:

- Zero Knowledge Tag: `zk`
- Zero Knowledge Handle: `zero-knowledge` e.g. /community/zero-knowledge 

Please provide your input on how to make this less confusing. Perhaps we need to rename handle to URL slug. Or maybe just keep labels for now.

File: src/community.rs (hunk 1 above)
Comment (akaia-shadowfox) [reply src/community.rs:23]: AFAIK we don't ( CC @ailisp )

Comment (ailisp) [reply src/community.rs:23]: No we don't, it's still okay to adjust this struct without worrying about migration

**Comment by ailisp** [comment]
Close as superseded by #31

File: src/community.rs (hunk 2 above)
Comment (frol) [reply src/community.rs:22]: @ori-near Why do we need those values to be different? If you want to use `zero-knowledge` in the URL, let's just apply `zero-knowledge` label to the posts where it is missing (either replace `zk` with it, or just add another label to the posts), and use `zero-knowledge` to filter the posts.

Comment (akaia-shadowfox) [reply src/community.rs:22]: @frol handles can be converted into account names later. Or is it okay to use accountId as a tag as well? 🤔

**Pull request closed without merging**